providers/universal-provider/test/index.spec.ts

```diff
@@ -316,6 +315,48 @@ describe("UniversalProvider", function () {
 
         await deleteProviders({ A: dapp, B: wallet });
       });
+      it("should automatically assign namespaces to optionalNamespaces", async () => {
+        const dapp = await UniversalProvider.init({
+          ...TEST_PROVIDER_OPTS,
+          name: "dapp",
+        });
+        const wallet = await UniversalProvider.init({
+          ...TEST_PROVIDER_OPTS,
+          name: "wallet",
+        });
+
+        const requestedRequiredNamespaces = {
+          eip155: {
+            chains: ["eip155:1", "eip155:2"],
+            methods: ["eth_sendTransaction", "eth_sign"],
+            events: ["chainChanged", "accountsChanged"],
+          },
+        };
```

Comment:
@ganchoradkov maybe overthinking but can we cover the sorting specifically in a test?

Test case: What happens of there is Solana mainnet in `requiredNamespaces` and dev/textnet in op`optional`.

Let's add a test that after this `mainnet` would be the first in the array.